nmigen_boards/genesys2.py

```diff
@@ -0,0 +1,390 @@
+                           Attrs(IOSTANDARD="LVCMOS18")),
+                 Subsignal("oled_vbat", Pins("AB22", dir="o"),
+                           Attrs(IOSTANDARD="LVCMOS33"))),
+        Resource("hdmi", 0,
+                 Subsignal("tx_clk",
+                           DiffPairs(p="AA20", n="AB20", dir="o"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("tx",
+                           DiffPairs(p="AC20 AA22 AB24",
+                                     n="AC21 AA23 AC25", dir="o"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("tx_scl", Pins("AF27", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("tx_sda", Pins("AF26", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("rx_clk",
+                           DiffPairs(p="AE28", n="AF28", dir="i")),
+                 Subsignal("rx",
+                           DiffPairs(p="AJ26 AG27 AH26",
+                                     n="AK26 AG28 AH27", dir="i"),
+                           Attrs(IOSTANDARD="TMDS_33")),
+                 Subsignal("rx_scl", Pins("AJ28", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33")),
+                 Subsignal("rx_sda", Pins("AJ29", dir="io"),
+                           Attrs(IOSTANDARD="LVCMOS33"))),
+        Resource("vga", 0,
+                 Subsignal("r", Pins("AK25 AG25 AH25 AK24 AJ24", dir="o")),
+                 Subsignal("g", Pins("AJ23 AJ22 AH22 AK21 AJ21 AK23",
+                                     dir="o")),
+                 Subsignal("b", Pins("AH20 AG20 AF21 AK20 AG22", dir="o")),
+                 Subsignal("hsync", PinsN("AF20", dir="o")),
+                 Subsignal("vsync", PinsN("AG23", dir="o")),
+                 Attrs(IOSTANDARD="LVCMOS33")),
+        *SDCardResources(0, clk="R28", cmd="R29", dat0="R26", dat1="R30",
+                        dat2="P29", dat3="T30", cd="P28",
+                        attrs=Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("sd_card_reset", 0,
+                 Pins("AE24", dir="o"), Attrs(IOSTANDARD="LVCMOS33")),
+        Resource("usb_otg", 0,
+                 Subsignal("d", Pins("AE14 AE15 AC15 AC16"
+                                     "AB15 AA15 AD14 AC14", dir="io")),
+                 Subsignal("clk", Pins("AD18", dir="i")),
+                 Subsignal("dir", Pins("Y16", dir="i")),
+                 Subsignal("stp", Pins("AA17", dir="o")),
+                 Subsignal("nxt", Pins("AA16", dir="i")),
+                 Subsignal("vbusoc", PinsN("AF16", dir="i")),
+                 Subsignal("reset", PinsN("AB14", dir="o")),
+                 Attrs(IOSTANDARD="LVCMOS18")),
+        Resource("eth_clocks", 0,
+                 Subsignal("tx_clk", Pins("AE10", dir="o")),
+                 Subsignal("rx_clk", Pins("AG10", dir="i")),
+                 Attrs(IOSTANDARD="LVCMOS18")),
+        Resource("eth", 0,
```

Comment:
This looks like a specific kind of Ethernet interface (RGMII?)

Take a look at the Atlys board to see how an RGMII interface should look like.

Reply:
`eth` →`eth_rgmii`, subsignals as per `AtlysPlatform`.